src/AuthorizedSocketConnection.js

```diff
@@ -158,48 +171,68 @@ export default class AuthorizedSocketConnection<TContext, TCredentials> {
         return;
       }
 
-      const subscriptionPromise = subscribe({
-        schema: this.config.schema,
-        document: documentAST,
-        variableValues: variables,
-        contextValue: {
-          ...this.config.context,
-          subscribe: (...args) => {
-            const source = this.config.subscriber.subscribe(...args);
-            return AsyncUtils.filter(source, this.isAuthorized);
-          },
+      const sourcePromise = createSourceEventStream(
+        this.config.schema,
+        document,
+        null,
+        {
+          subscribe: (...args) =>
+            AsyncUtils.filter(
+              this.config.subscriber.subscribe(...args),
+              this.isAuthorized,
+            ),
         },
-      });
+        variables,
+      );
 
-      this.subscriptions.set(id, subscriptionPromise);
+      this.subscriptions.set(id, sourcePromise);
 
       try {
-        result = await subscriptionPromise;
+        resultOrStream = await sourcePromise;
       } catch (err) {
         this.subscriptions.delete(id);
         throw err;
       }
 
-      if (result.errors != null) {
+      if (resultOrStream.errors != null) {
         this.subscriptions.delete(id);
         this.emitError({
           code: 'subscribe_failed.gql_error',
           // $FlowFixMe
-          data: result.errors,
+          data: resultOrStream.errors,
         });
+
         return;
       }
     } finally {
       acknowledge(cb);
     }
 
-    const subscription: AsyncIterable<ExecutionResult> = (result: any);
-    for await (const payload of subscription) {
-      this.log('info', 'emit', {
-        payload,
-        credentials: this.getCredentials(),
-      });
-      this.config.socket.emit('subscription update', { id, ...payload });
+    const stream: AsyncIterable<mixed> = (resultOrStream: any);
+
+    for await (const payload of stream) {
+      const credentials = this.config.credentialsManager.getCredentials();
+
+      let response;
+      try {
+        response = await execute(
+          this.config.schema,
+          document,
+          payload,
+          this.config.createContext && this.config.createContext(credentials),
+          variables,
+        );
+      } catch (e) {
+        if (e instanceof GraphQLError) {
+          response = { errors: [e] };
+        } else {
+          throw e;
+        }
+      }
+
+      this.log('info', 'emit', { response, credentials });
```

Comment:
this is a change to the log message... `response` instead of `payload`. is this a problem? i'm trying to be consistent w/naming within `graphql/subscribe` cc @yodaiken